src/components/AnonymousChat/child/ChatReplyPreview.tsx

```diff
@@ -0,0 +1,111 @@
+import Animated, {SlideInDown, SlideOutDown} from 'react-native-reanimated';
+import {StyleSheet, Text, TouchableOpacity, View} from 'react-native';
+
+import IconClear from '../../../assets/icon/IconClear';
+import dimen from '../../../utils/dimen';
+import useMessageHook from '../../../hooks/screen/useMessageHook';
+import {colors} from '../../../utils/colors';
+import {fonts, normalizeFontSize} from '../../../utils/fonts';
+
+interface ChatReplyPreviewProps {
+  type: 'SIGNED' | 'ANONYMOUS';
+}
+
+const ChatReplyPreview = ({type}: ChatReplyPreviewProps) => {
+  const {replyPreview, clearReplyPreview} = useMessageHook();
+
+  const textContainerStyle = [
+    type === 'SIGNED' ? styles.containerSigned : styles.containerAnon,
+    styles.textContainer
+  ];
+
+  return (
+    <Animated.View
+      entering={SlideInDown.duration(150)}
+      exiting={SlideOutDown.duration(80)}
+      style={styles.container}>
+      <View style={textContainerStyle}>
+        <View style={styles.chatTitleContainer}>
+          <Text style={styles.userText}>{replyPreview?.username}</Text>
+          <View style={styles.dot} />
+          <Text style={styles.timeText}>{replyPreview?.time}</Text>
+        </View>
+
+        <Text style={styles.text} numberOfLines={1}>
+          {`${replyPreview?.message}`}
+        </Text>
+      </View>
+
+      <TouchableOpacity
+        onPress={clearReplyPreview}
+        activeOpacity={0.75}
+        style={styles.containerDismiss}>
+        <IconClear fill={colors.black} width={12} height={12} />
+      </TouchableOpacity>
+    </Animated.View>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    flexDirection: 'row',
+    paddingVertical: dimen.normalizeDimen(8),
+    paddingLeft: dimen.normalizeDimen(8),
+    paddingRight: dimen.normalizeDimen(5),
+    alignItems: 'center',
+    backgroundColor: colors.lightgrey
+  },
+  containerSigned: {
+    backgroundColor: colors.darkBlue
+  },
+  containerAnon: {
+    backgroundColor: colors.anon_primary
+  },
+  textContainer: {
+    flex: 1,
+    paddingVertical: dimen.normalizeDimen(4),
+    paddingHorizontal: dimen.normalizeDimen(8),
+    borderRadius: 8
+  },
+  chatTitleContainer: {
+    display: 'flex',
+    flexDirection: 'row'
+  },
+  userText: {
+    fontFamily: fonts.inter[600],
+    fontSize: 12,
+    lineHeight: 19.36,
+    color: colors.white
+  },
+  dot: {
+    width: 3,
+    height: 3,
+    borderRadius: 2,
+    marginLeft: 5,
+    marginRight: 5,
+    backgroundColor: colors.white,
+    alignSelf: 'center'
+  },
+  timeText: {
+    fontFamily: fonts.inter[200],
+    fontSize: 10,
+    lineHeight: 12.19,
+    alignSelf: 'center',
+    color: colors.white
+  },
+  text: {
+    fontFamily: fonts.inter[400],
+    fontSize: normalizeFontSize(16),
+    lineHeight: 19.36,
+    marginBottom: 4,
+    color: colors.white
+  },
+  containerDismiss: {
+    marginLeft: 5,
+    padding: 5
+  }
+});
+
+export default React.memo(ChatReplyPreview);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `ChatReplyPreview` component seems to be well implemented. It uses the `useMessageHook` custom hook to get the reply preview and clear it when necessary. The component also handles animations for entering and exiting using `react-native-reanimated`. The styling is done using `StyleSheet.create`, which is a good practice in React Native for performance reasons. However, I would suggest adding PropTypes or TypeScript interfaces for better type checking and self-documentation.

```diff
+ import PropTypes from 'prop-types';

  interface ChatReplyPreviewProps {
    type: 'SIGNED' | 'ANONYMOUS';
  }

+ ChatReplyPreview.propTypes = {
+   type: PropTypes.oneOf(['SIGNED', 'ANONYMOUS']).isRequired,
+ };
```

Also, consider handling the case where `replyPreview` is `null` or `undefined`. Currently, if `replyPreview` is `null`, it might cause issues when trying to access its properties like `username`, `time`, and `message`.

```diff
  return (
    <Animated.View
      entering={SlideInDown.duration(150)}
      exiting={SlideOutDown.duration(80)}
      style={styles.container}>
      {replyPreview && (
        <>
          <View style={textContainerStyle}>
            <View style={styles.chatTitleContainer}>
              <Text style={styles.userText}>{replyPreview.username}</Text>
              <View style={styles.dot} />
              <Text style={styles.timeText}>{replyPreview.time}</Text>
            </View>

            <Text style={styles.text} numberOfLines={1}>
              {`${replyPreview.message}`}
            </Text>
          </View>

          <TouchableOpacity
            onPress={clearReplyPreview}
            activeOpacity={0.75}
            style={styles.containerDismiss}>
            <IconClear fill={colors.black} width={12} height={12} />
          </TouchableOpacity>
        </>
      )}
    </Animated.View>
  );
```
This way, the component will not try to render the reply preview if it is not available.


<!-- This is an auto-generated comment by OSS CodeRabbit -->